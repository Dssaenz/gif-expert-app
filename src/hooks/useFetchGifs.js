import { useEffect, useState } from "react"
import getCategories from "../helpers/getGifs";

const useFetchGifs = (categories) => {
	const [ state, setState ] = useState({
		data: [],
		loading: true,
	})

	useEffect(() => {
		getCategories(categories).then(items => setState({
			data: items,
			loading: false
		}))
	}, [categories])

	return state;
}

export default useFetchGifs