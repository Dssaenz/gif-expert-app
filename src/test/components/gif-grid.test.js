import React from 'react'
import '@testing-library/jest-dom';
import Gifgrid from '../../components/gif-grid'
import { shallow } from 'enzyme';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test in <GifGrid /> component', () => {
	const categories = '';

	test('It should display the component correctly', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		})

		const wrapper = shallow(<Gifgrid categories={categories} />)
		expect( wrapper ).toMatchSnapshot();
	})


	test('Should display items when images useFetchGifs is fill', () => {

		const gifs = [{
			id: '123',
			title: 'https://localhost/cosa.png',
			url: 'una cosa'
		}]
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		})

		const wrapper = shallow(<Gifgrid categories={categories} />)

		// Exists 'h1' element
		expect( wrapper.find('h1').exists() ).toBe(false);

		// Number of the element
		expect( wrapper.find('GifGridItem').length).toBe(gifs.length);
	})
})