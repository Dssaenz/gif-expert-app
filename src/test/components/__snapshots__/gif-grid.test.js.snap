// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid /> component It should display the component correctly 1`] = `
<h1>
  Cargando...
</h1>
`;
